app/src/main/cpp/BrowserWorld.cpp

```diff
@@ -1178,6 +1178,29 @@ BrowserWorld::EndFrame() {
   VRBrowser::HandleAudioPose(q.x(), q.y(), q.z(), q.w(), p.x(), p.y(), p.z());
 }
 
+void
+BrowserWorld::TriggerHapticFeedback(const float aPulseDuration, const float aPulseIntensity) {
+  if (!m.controllers) {
+    return;
+  }
+
+  bool focused = false;
+  int32_t index = 0;
+  uint64_t inputFrameID = 0;
+  for (Controller& controller: m.controllers->GetControllers()) {
+    focused = controller.focused;
+    if (focused) {
+      index = controller.index;
+      inputFrameID = controller.inputFrameID + 1;
+      break;
+    }
+  }
+
+  if (focused && m.controllers->GetHapticCount(index)) {
+    m.controllers->SetHapticFeedback(index, inputFrameID, aPulseDuration, aPulseIntensity);
```

Comment:
what about doing this inside the loop? We could get rid of the `focused` variable and replace the break by a return. Also the index and inputFrameID won't be needed.

Reply:
Done